app/Domain/Clients/Controllers/ShowClient.php

```diff
@@ -52,38 +54,95 @@ public function init(
+            }
+
+            $clientValues = (object) [
+                'id' => $row->id,
+                'name' => $row->name,
+                'street' => $row->street,
+                'zip' => $row->zip,
+                'city' => $row->city,
+                'state' => $row->state,
+                'country' => $row->country,
+                'phone' => $row->phone,
+                'internet' => $row->internet,
+                'email' => $row->email,
+            ];
+
+
+            if (empty($row) === false && Auth::userIsAtLeast(Roles::$admin)) {
+                $file = app()->make(FileRepository::class);
+                $project = app()->make(ProjectRepository::class);
+
+                if (session('userdata.role') == 'admin') {
+                    $this->tpl->assign('admin', true);
+                }
+
+                $this->tpl->assign('userClients', $this->clientService->getUserClients($id));
+                $this->tpl->assign('comments', $this->commentService->getComments('client', $id));
+                $this->tpl->assign('imgExtensions', ['jpg', 'jpeg', 'png', 'gif', 'psd', 'bmp', 'tif', 'thm', 'yuv']);
+                $this->tpl->assign('client', $clientValues);
+                $this->tpl->assign('users', app()->make(UserRepository::class));
+                $this->tpl->assign('clientProjects', $project->getClientProjects($id));
+                $this->tpl->assign('files', $file->getFilesByModule('client', $id));
+
+                return $this->tpl->display('clients.showClient');
+            } else {
+                return $this->tpl->display('errors.error403');
+            }
+        }
+
+
         /**
-         * run - display template and edit data
+         * post - display template and edit data
          */
-        public function run()
+        public function post($params)
         {
+
             Auth::authOrRedirect([Roles::$owner, Roles::$admin, Roles::$manager], true);
 
             $id = '';
 
-            if (isset($_GET['id']) === true) {
-                $id = (int) ($_GET['id']);
+            if (isset($params['id']) === true) {
+                $id = (int) ($params['id']);
             }
 
-            $row = $this->clientRepo->getClient($id);
+            $row = $this->clientService->get($id);
 
             if ($row === false) {
                 $this->tpl->display('errors.error404');
 
-                return;
+                return $this->tpl->display('errors.error404');
             }
 
-            $clientValues = [
-                'id' => $row['id'],
-                'name' => $row['name'],
-                'street' => $row['street'],
-                'zip' => $row['zip'],
-                'city' => $row['city'],
-                'state' => $row['state'],
-                'country' => $row['country'],
-                'phone' => $row['phone'],
-                'internet' => $row['internet'],
-                'email' => $row['email'],
+            $clientValues = (object) [
```

Comment:
The clientService should already return an object/model. This mapping should not be needed anymore.